packages/inline-edit/src/InlineEdit.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {
+  FC, createRef, useState, useEffect, useCallback, memo,
+} from 'react';
+import { withTheme } from 'styled-components';
+
+
+import { getInlineEditUncontrolled } from './InlineEditUncontrolled';
+import { IInlineEditProps } from './interfaces';
+
+
+interface IInlineEditState {
+  isEditing?: boolean;
+}
+
+const getPureInlineEdit: <TFieldValue = string>() => FC<
+IInlineEditProps<TFieldValue> & IInlineEditState> = function f<TFieldValue>() {
+  return ({
+    startWithEditViewOpen = false,
+    onConfirm,
+    defaultValue,
+    readView,
+    editView,
+    ...rest
+  }: IInlineEditProps<TFieldValue>) => {
+    const editViewRef = createRef<HTMLElement>();
+    const [isEditing, setIsEditing] = useState(startWithEditViewOpen);
+
+    useEffect(() => {
+      if (startWithEditViewOpen && editViewRef.current) {
+        editViewRef.current.focus();
+      }
+    }, [startWithEditViewOpen, editViewRef]);
+
+    useEffect(() => {
+      if (isEditing && editViewRef.current) {
+        editViewRef.current.focus();
+      }
+    }, [isEditing, editViewRef]);
+
+    const handleConfirm = useCallback((value: string): void => {
+      setIsEditing(false);
+      onConfirm(value);
+    }, [onConfirm]);
+
+    const handleCancel = useCallback((): void => {
+      setIsEditing(false);
+      onConfirm('');
```

Comment:
Why you call onConfirm instead of onCancel?

Reply:
It is problem with understanding, what we should do in case: if we confirm value first, then we input new value, should we set value to empty string (onConfirm('')) or reset to last value (onConfirm(lastValue)) after clicking 'cancel' button? So, I think it is temporary decision.